src/main/scala/diplomacy/Resources.scala

```diff
@@ -392,6 +394,10 @@ object BindingScope
     case x: BindingScope => find(x.parent).orElse(Some(x))
     case x => find(x.parent)
   }
+
+  var bindingScopes = new collection.mutable.ArrayBuffer[BindingScope]()
```

Comment:
Why is it both an `ArrayBuffer` and a `var`? 

This should probably be a `private val`